static/js/components/dashboard/CourseAction_test.js

```diff
@@ -63,11 +63,13 @@ describe('CourseAction', () => {
     const wrapper = shallow(<CourseAction course={course} now={now} checkout={checkoutStub}/>);
     let buttonContainer = wrapper.find(".course-action-action");
     let description = wrapper.find(".course-action-description");
-    let buttonProps = buttonContainer.find("button").props();
+    let buttonProps = buttonContainer.find(".dashboard-button").props();
+    let spanText = buttonContainer.find(".sr-only").text();
+    let buttonText = buttonContainer.find(".dashboard-button").children().text();
     let firstRun = course.runs[0];
 
     assert(!buttonProps.disabled);
-    assert.equal(buttonContainer.text(), `Upgrade in ${firstRun.title}`);
+    assert.equal(`${buttonText}${spanText}`, `Upgrade in ${firstRun.title}`);
```

Comment:
@gsidebo 

2. Text is changing  buttonContainer.text() === (Enroll + course title) or (upgrade + course title) or (empty string)